src/details/ArborX_AccessTraits.hpp

```diff
@@ -103,6 +103,26 @@ struct AccessTraitsHelper<AccessTraits<X, Tag>>
   using tag = Kokkos::detected_t<PredicateTagArchetypeAlias, type>;
 };
 
+template <typename X, typename Tag>
+class Adapt
+{
+  X _x;
+  using Access = AccessTraits<X, Tag>;
+
+public:
+  Adapt(Tag, X const &x)
+      : _x(x)
+  {}
+  using memory_space = typename Access::memory_space;
+  using value_type = std::decay_t<
+      Kokkos::detected_t<AccessTraitsGetArchetypeExpression, Access, X>>;
+  using size_type =
+      Kokkos::detected_t<AccessTraitsSizeArchetypeExpression, Access, X>;
+  using tag = Kokkos::detected_t<PredicateTagArchetypeAlias, value_type>;
```

Comment:
Guess this class is almost like a [`RangeAdaptor`](https://github.com/aprokop/ArborX/blob/api_v2_t2/src/details/ArborX_RangeAdaptor.hpp), except for this tag.

Reply:
I will remove it.  It was mostly an intermediate state.  It can be deduced from `value_type`